frontend/src/metabase/dashboard/hoc/controls/hooks/use-dashboard-fullscreen.ts

```diff
@@ -0,0 +1,56 @@
+import { useState } from "react";
+import { useMount, useUnmount } from "react-use";
+import screenfull from "screenfull";
+
+import type { DashboardFullscreenControls } from "metabase/dashboard/hoc/controls/types";
+
+export const useDashboardFullscreen = (): DashboardFullscreenControls => {
+  const [isFullscreen, setIsFullscreen] = useState(false);
+
+  const _fullScreenChanged = () => {
+    setIsFullscreen(screenfull.isFullscreen);
+  };
+
+  useMount(() => {
+    if (screenfull.isEnabled) {
+      document.addEventListener(
+        screenfull.raw.fullscreenchange,
+        _fullScreenChanged,
+      );
+    }
+  });
+
+  useUnmount(() => {
+    if (screenfull.isEnabled) {
+      document.removeEventListener(
+        screenfull.raw.fullscreenchange,
+        _fullScreenChanged,
+      );
+    }
+  });
+
+  const onFullscreenChange = async (
+    newIsFullscreen: boolean,
+    browserFullscreen: boolean = true,
+  ) => {
+    if (newIsFullscreen !== isFullscreen) {
```

Comment:
Basically in the previous code, `newIsFullscreen` was also named `isFullscreen` while there was also a state variable called `isFullscreen`. So `newIsFullscreen` is supposed to be the next value.

For `browserFullscreen` - that's supposed to be the native fullscreen.

Maybe we can change the names here then:
- `newIsFullscreen` => `nextIsFullscreen` or something like that
- `browserFullscreen` => `shouldRequestNativeFullscreen`

Reply:
I did a thing and used the mantine `useFullscreen` hook instead of `screenfull`, which handles a lot more of the logic for us. It helps get rid of the nesting